pkg/etcd/etcd.go

```diff
@@ -329,6 +329,16 @@ func (e *ETCD) join(ctx context.Context, clientAccessInfo *clientaccess.Info) er
 	}
 
 	for _, member := range members.Members {
+		lastHyphen := strings.LastIndex(member.Name, "-")
+		memberNodeName := member.Name[:lastHyphen]
+		if memberNodeName == e.config.ServerNodeName {
+			// make sure to remove the name file if a duplicate node name is used
+			nameFile := nameFile(e.config)
+			if err := os.Remove(nameFile); err != nil {
+				return err
+			}
+			return errors.New("Failed to join etcd cluster due to duplicate node names, please use unique node name for the server")
```

Comment:
Talking through the logic here, when joining an etcd cluster, if we find an etcd member that has the same node name as the existing node name we're trying to join with, we remove the namefile we've generated and return an error so that the user can change the hostname/nodename and can then join the node, right? 

What if there is an orphaned etcd member that is still part of the cluster that matches the existing node name (or the existing node is attempting to replace a broken node), this seems to be problematic as there would be no way to clean up after the old etcd node/member without deleting the v1.Node object or annotating it with a removal annotation that corresponds to this node first... which we may not want to do.

I wonder if we should make the error message that is returned have a higher fidelity, so that the user can more intelligently make a decision on how to recover.

Reply:
yes to the first question, I do remove the name file so that the user can change the name and be used in the cluster, for the orphaned member part, I think erroring out is better in this case than just allowing it to join the cluster, for a scenario, where you have:
- server1 2 and 3
- you remove node 3 and some how it doesnt get removed from the member list
- you try to rejoin node3, now you get an error and this can alert the user that he has some orphaned node

I can try to make the error message more indicative that there might be a problem with orphaned nodes